packages/components/src/date-time/style.scss

```diff
@@ -54,19 +54,35 @@
 		font-size: $default-font-size;
 	}
 
+	// Seperate borders so that border respect border radius
+	.CalendarMonth_table {
+		border-collapse: separate;
+		border-spacing: 2px;
+	}
+
 	.CalendarDay {
 		font-size: $default-font-size;
-		border: 1px solid transparent;
+		border: none;
 		border-radius: $radius-round;
 		text-align: center;
+
+		&:focus {
+			box-shadow: inset 0 0 0 $border-width-focus theme(primary), inset 0 0 0 #{ $border-width-focus + $border-width } $white;
```

Comment:
The focus style is drawn using box shadows to create a circle. There's an inset white lower layer shadow to make it visible even on a selected day.

Windows 10 high contrast mode gets a transparent outline instead, which it makes opaque, and it strips away the box shadow.